File: Sample/java/src/main/java/net/nend/sample/java/fullboard/FullBoardWebViewActivity.java
package net.nend.sample.java.fullboard;

import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.util.Log;

import androidx.appcompat.app.AppCompatActivity;

import net.nend.android.NendAdFullBoard;
import net.nend.android.NendAdFullBoardLoader;
import net.nend.sample.java.R;

public class FullBoardWebViewActivity extends AppCompatActivity implements NendAdFullBoard.FullBoardAdListener {

    private NendAdFullBoardLoader mLoader;
    private NendAdFullBoard mAd;
    private Handler mHandler = new Handler(Looper.getMainLooper());

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_full_board_web);

        final ObservableWebView webView = findViewById(R.id.webview);
        webView.loadUrl("https://board.nend.net");
        webView.setCallback(new ObservableWebView.Callback() {
            @Override
            public void onScrollEnd() {
                if (mAd != null) {
                    mHandler.postDelayed(new Runnable() {
                        @Override
                        public void run() {
                            mAd.show(FullBoardWebViewActivity.this);
                        }
                    }, 500L);
                }
            }
        });

        mLoader = new NendAdFullBoardLoader(getApplicationContext(), 485520, "a88c0bcaa2646c4ef8b2b656fd38d6785762f2ff");
        loadAd();
    }

    private void loadAd() {
        mLoader.loadAd(new NendAdFullBoardLoader.Callback() {
            @Override
            public void onSuccess(NendAdFullBoard nendAdFullBoard) {
                mAd = nendAdFullBoard;
                mAd.setAdListener(FullBoardWebViewActivity.this);
            }

            @Override
            public void onFailure(NendAdFullBoardLoader.FullBoardAdError fullBoardAdError) {
                Log.w("FullBoard", fullBoardAdError.name());
            }
        });
    }

    @Override
    public void onDismissAd(NendAdFullBoard nendAdFullBoard) {
        loadAd();
    }

    @Override
    public void onShowAd(NendAdFullBoard nendAdFullBoard) {

    }

    @Override
    public void onClickAd(NendAdFullBoard nendAdFullBoard) {

    }
}

File: Sample/kotlin/src/main/java/net/nend/sample/kotlin/fullboard/FullBoardWebViewActivity.kt
package net.nend.sample.kotlin.fullboard

import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import android.util.Log
import kotlinx.android.synthetic.main.activity_full_board_web.*
import net.nend.android.NendAdFullBoard
import net.nend.android.NendAdFullBoardLoader
import net.nend.sample.kotlin.R
import net.nend.sample.kotlin.fullboard.FullBoardMenuActivity.Companion.FULLBOARD_API_KEY
import net.nend.sample.kotlin.fullboard.FullBoardMenuActivity.Companion.FULLBOARD_LOG_TAG
import net.nend.sample.kotlin.fullboard.FullBoardMenuActivity.Companion.FULLBOARD_SPOT_ID

class FullBoardWebViewActivity : AppCompatActivity(), NendAdFullBoard.FullBoardAdListener {

    private lateinit var loader: NendAdFullBoardLoader
    private var ad: NendAdFullBoard? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_full_board_web)

        webview.loadUrl("https://board.nend.net")
        webview.setCallback(object : ObservableWebView.Callback {
            override fun onScrollEnd() {
                ad?.let {
                    Handler().postDelayed({ it.show(this@FullBoardWebViewActivity) }, 500L)
                }
            }
        })

        loader = NendAdFullBoardLoader(applicationContext, FULLBOARD_SPOT_ID, FULLBOARD_API_KEY)
        loadAd()
    }

    private fun loadAd() {
        loader.loadAd(object : NendAdFullBoardLoader.Callback {
            override fun onSuccess(nendAdFullBoard: NendAdFullBoard) {
                ad = nendAdFullBoard
                ad?.setAdListener(this@FullBoardWebViewActivity)
            }

            override fun onFailure(fullBoardAdError: NendAdFullBoardLoader.FullBoardAdError) {
                Log.w(FULLBOARD_LOG_TAG, fullBoardAdError.name)
            }
        })
    }

    override fun onDismissAd(nendAdFullBoard: NendAdFullBoard) {
        loadAd()
    }

    override fun onShowAd(nendAdFullBoard: NendAdFullBoard) {

    }

    override fun onClickAd(nendAdFullBoard: NendAdFullBoard) {

    }
}